### PR TITLE
Abdul Session6 Assignment

### PR DESCRIPTION
**Pull request by Abdul-Andha**
- Unit tests:
  - Created 3 unit tests for `InfoForm` component, including happy and sad
    - Tested the form loads
    - Tested the form throws an error when parameters aren't sent
    - Tested the form fields fill out as expected
  - Created 4 unit tests for `MetricsForm` component, including happy and sad
    - Tested the form loads
    - Tested the form throws an error when parameters aren't sent
    - Tested the form fields fill out as expected
    - Tested the weight field does not fill out when typing letters
  - Increased test coverage for `app/signup/_components` (where the two components are) from 0% to 100% for statements, branches, functions, and lines

- Integration tests:
  - Created 5 integration tests for `Signup`, which combines `InfoForm` and `MetricsForm` components, including happy and sad
    - Tested info form loads first
    - Tested metrics form does not load initially
    - Tested metrics form loads up when info form is filled out and next is pressed
    - Tested metrics form does not load up when not all fields in info form are filled out and next is pressed
    - Tested signup action is called when both forms are filled out correctly and create account is clicked
  - Increased test coverage for `app/signup` from 0% to 71.55% statements, 75% branches, 80% functions, and 71.55% lines

- End to End tests:
  - Created 1 e2e test for user creating a new account flow
    - User starts at home page, navigates to signup for, fills out InfoForm and clicks next, fills out MetricsForm and clicks create account, and is redirected to home page on successful signup.

### CODE REVIEW

File: topics-project/__tests__/signup.test.js [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+import Signup from '../app/signup/page'
+import { InfoForm } from '../app/signup/_components/infoForm'
+import { MetricsForm } from '../app/signup/_components/metricsForm'
+import * as actions from '../app/signup/actions'
+import { expect } from '@playwright/test';
+
+jest.mock('../app/signup/actions')
+
+// Unit tests for info form component
+describe('Info Form', () => {
```
Comment (navteniev) [inline topics-project/__tests__/signup.test.js:14]: the standard approach is to separate the tests for individual components into independent files. Which would have make this change a bit simpler to understand. Having all the tests in a single file like this makes it difficult to understand test failures.

File: topics-project/__tests__/signup.test.js [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    expect(weightField.value).toEqual('');
+  })
+
+});
+
+// Integration tests for signup
+// Test action to create new user
+// Check if new user exists in db using /get-user endpoint
+
+describe('Signup integration tests', () => {
```
Comment (navteniev) [inline topics-project/__tests__/signup.test.js:159]: It's not clear for this test that is actually does what the comment explains.
What would have made this test clearer is if the user data was retrieved from the data store.

For example completing the sign up and then calling the data store in the test to confirm that the use is present.

This test relies on the UI to function correctly to verify the user has been created.

Additionally if this is working as expected then it's creating user in the datastore but not deleting them. 

How would this test run if you run it multiple times?

**Review by navteniev — COMMENTED** [review]
Unit test: 2/2
Integration test: 2/2
end-to-end test: 1/1
CI: 4/4
PR 1/1

Final 10/10 pts

File: topics-project/__tests__/signup.test.js (hunk 2 above)
Comment (Abdul-Andha) [reply topics-project/__tests__/signup.test.js:159]: Got it, will be fixed before we push this test to main!

**Pull request closed without merging**